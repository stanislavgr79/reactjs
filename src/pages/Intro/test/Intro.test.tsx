import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

function Intro(props: { message: string | undefined, time: string | undefined }) {
  return (
    <>
      <div className="message">
        <h1>{props.message}</h1>
      </div>
      <div id="time">
        <h1>{props.time}</h1>
      </div>
    </>
  );
}

const component = TestRenderer.create(<Intro message="task2" time="2:31:55 PM"></Intro>);

describe('Intro', () => {
  test('snapshot renders', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test('props should be find', () => {
  render(<Intro message="task2" time="2:31:55 PM"></Intro>);
  const message = screen.getByText(/task2/i);
  const time = screen.getByText(/2:31:55 PM/i);
  expect(message).toBeInTheDocument();
  expect(time).toBeInTheDocument();
});
