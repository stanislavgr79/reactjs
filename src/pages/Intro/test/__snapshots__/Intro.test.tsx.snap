// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro snapshot renders 1`] = `
Array [
  <div
    className="message"
  >
    <h1>
      task2
    </h1>
  </div>,
  <div
    id="time"
  >
    <h1>
      2:31:55 PM
    </h1>
  </div>,
]
`;
